src/app/features/task-details/task-details.component.ts

```diff
@@ -36,57 +44,43 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
     assignedTo: ['', [Validators.required]],
   });
   task: TaskModel = <TaskModel>{};
-  private destroy: Subject<void> = new Subject();
   errorMessage: string = '';
   editMode: boolean = false;
 
   constructor(
     protected userService: UserService,
     private fb: FormBuilder,
-    private taskService: TaskService,
     private route: ActivatedRoute,
-    private languageService: LanguageService
+    private store: Store<AppState>
   ) {
   }
 
   ngOnInit(): void {
     const taskId: string | null = this.route.snapshot.paramMap.get('id');
     if (taskId) {
-      const fetchedTask: Observable<TaskModel> = this.taskService.getTask(taskId);
-      if (fetchedTask) {
-        fetchedTask
-          .pipe(takeUntil(this.destroy))
-          .subscribe((res: TaskModel) => {
-            this.task = res;
-            this.currentTask.setValue({
-              title: this.task.title,
-              description: this.task.description,
-              type: this.task.type,
-              status: this.task.status,
-              assignedTo: this.task.assignedTo,
-            });
+      this.store.select(selectOpenedTask)
+        .pipe(takeUntil(this.destroy))
```

Comment:
Also I noticed that in 99% of use cases fr subscriptions yo uuse takeUntil, but sometimes think if it is needed to subscribe to this variable all the time, maybe you need to subscribe only once, because you only need the value once on page load, maybe you can async it through the template, this would help you improve performance of your app